spec/requests/api/graphql/mutations/add_user_to_submission_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "support/gravity_helper"
+
+describe "addUserToSubmission mutation" do
+  let(:user) { Fabricate(:user, gravity_user_id: "userid") }
+
+  let(:token) do
+    payload = {aud: "gravity", sub: user.gravity_user_id, roles: "user"}
+    JWT.encode(payload, Convection.config.jwt_secret)
+  end
+
+  let(:headers) { {"Authorization" => "Bearer #{token}"} }
+
+  before do
+    allow(Convection.config).to receive(:send_new_receipt_email).and_return(true)
+  end
+
+  it "associates a user with an unclaimed draft submission" do
+    submission = Fabricate(:submission, user: nil)
+
+    mutation = <<-GRAPHQL
+        mutation {
+          addUserToSubmission(input: { id: \"#{submission.id}\" }) {
+            clientMutationId
+            consignmentSubmission {
+              internalID
+            }
+          }
+        }
+    GRAPHQL
+
+    expect(SubmissionService).to receive(:create_my_collection_artwork).and_return(true)
+
+    post "/api/graphql", params: {query: mutation}, headers: headers
+
+    expect(response.status).to eq 200
+    body = JSON.parse(response.body)
+
+    expect(body["data"]["addUserToSubmission"]["consignmentSubmission"]["internalID"]).to eq submission.id.to_s
+  end
+
+  it "no-ops and returns the submission if the current user is already associated with the submission" do
+    submission = Fabricate(:submission, user: user)
+
+    mutation = <<-GRAPHQL
+        mutation {
+          addUserToSubmission(input: { id: \"#{submission.id}\" }) {
+            clientMutationId
+            consignmentSubmission {
+              internalID
+            }
+          }
+        }
+    GRAPHQL
+
```

Comment:
**suggestion**: I would add something like `expect(SubmissionService).to receive(:add_user_to_submission).never` to confirm that it actually no-ops